### ISSUE TITLE
how can retrieve multi-file from skipper-gridfs

### ISSUE THREAD

**Issue by hemedani** [balderdashy/skipper#130]
for getting 1 file from gridfs we should do this in sails.js:
```javascript
download: function (req, res) {
		var blobAdapter = require('skipper-gridfs')({
			uri: 'mongodb://localhost:27017/shahrara.fileB'
		});

		var fd = req.param('fd'); // value of fd comes here from get request
		blobAdapter.read(fd, function (error, file) {
			if (error) {
				res.json(error);
			} else {
				res.contentType(req.param('type'));
				res.send(new Buffer(file));
			}
		});
	}
```
think we have the model named file and save some property like file name and type, how can handle getting all(or multi) file(in fact image) from database ? 

```javascript
allpic: function (req, res, next) {
		var blobAdapter = require('skipper-gridfs')({
			uri: 'mongodb://localhost:27017/shahrara.fileB'
		});
		File.find({})
			.exec(function findCB(err, found) {

				while (found.length) {
					var fd = found.pop().fileName;
					var typo = found.pop().type;
					// console.log('Found File with name ' + typo);
					console.log('Found File with name ' + fd);

					blobAdapter.read(fd, function (error, file) {
						if (error) {
							res.json(error);
						} else {
							res.contentType(typo);
							res.send(new Buffer(file));
						}
					});
				}
```
with the above code we throw into this error:
```
_http_outgoing.js:344
    throw new Error('Can\'t set headers after they are sent.');
    ^

Error: Can't set headers after they are sent.
    at ServerResponse.OutgoingMessage.setHeader (_http_outgoing.js:344:11)
    at ServerResponse.res.setHeader (/home/amir/Work/Web/Nahaei/shahrAra/manabe/server/node_modules/sails/node_modules/connect/lib/patch.js:133:22)
    at ServerResponse.res.set.res.header (/home/amir/Work/Web/Nahaei/shahrAra/manabe/server/node_modules/sails/node_modules/express/lib/response.js:577:10)
    at ServerResponse.res.contentType.res.type (/home/amir/Work/Web/Nahaei/shahrAra/manabe/server/node_modules/sails/node_modules/express/lib/response.js:446:15)
    at /home/amir/Work/Web/Nahaei/shahrAra/manabe/server/api/controllers/FileController.js:118:12
    at /home/amir/Work/Web/Nahaei/shahrAra/manabe/server/node_modules/skipper-gridfs/index.js:91:32
    at ConcatStream.<anonymous> (/home/amir/Work/Web/Nahaei/shahrAra/manabe/server/node_modules/concat-stream/index.js:36:43)
    at emitNone (events.js:67:13)
    at ConcatStream.emit (events.js:166:7)
    at finishMaybe (/home/amir/Work/Web/Nahaei/shahrAra/manabe/server/node_modules/readable-stream/lib/_stream_writable.js:460:14)
    at afterWrite (/home/amir/Work/Web/Nahaei/shahrAra/manabe/server/node_modules/readable-stream/lib/_stream_writable.js:340:3)
    at /home/amir/Work/Web/Nahaei/shahrAra/manabe/server/node_modules/readable-stream/lib/_stream_writable.js:327:9
    at nextTickCallbackWith0Args (node.js:433:9)
    at process._tickDomainCallback (node.js:403:13)
```

**Comment by irlnathan** [comment]
Hey @hemedani, would you mind [adding information](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md#ii-issue-contributions) about your Sails, Node, and NPM versions to this issue?  Thanks!

**Comment by sailsbot** [comment]
Thanks for posting, @hemedani.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
